src/test/regress/input/tablespace.source

```diff
@@ -146,3 +146,57 @@ DROP SCHEMA testschema CASCADE;
 
 DROP ROLE tablespace_testuser1;
 DROP ROLE tablespace_testuser2;
+
+-- Create a new tablespace and check that within this tablespace,
+-- after dropping the table, the size of all deleted (but still opened) files is 0.
+CREATE TABLESPACE testspace LOCATION '@testtablespace@';
+DROP TABLE IF EXISTS t;
+
+-- Checking that the AO row-oriented table doesn't leave non-zero files.
+CREATE TABLE t(a int, b int) WITH (appendoptimized = true, orientation=row) TABLESPACE testspace;
+INSERT INTO t SELECT i, i FROM generate_series(1,1000) i;
+DROP TABLE t;
+\! find /proc/[0-9]*/fd -lname '@testtablespace@*(deleted)' -exec stat -Lc '%s' {} \; 2>/dev/null | awk 'BEGIN {sum=0} {sum += $1} END {print "size:",  sum}';
+
+-- Checking that the AO column-oriented table doesn't leave non-zero files.
+CREATE TABLE t(a int, b int) WITH (appendoptimized = true, orientation=column) TABLESPACE testspace;
+INSERT INTO t SELECT i, i FROM generate_series(1,1000) i;
+DROP TABLE t;
+\! find /proc/[0-9]*/fd -lname '@testtablespace@*(deleted)' -exec stat -Lc '%s' {} \; 2>/dev/null | awk 'BEGIN {sum=0} {sum += $1} END {print "size:",  sum}';
+
+-- start_ignore
+-- Enable gp_inject_fault extension.
+-- This extension will be used in test below to force flushing of all dirty
+-- pages by the bg writer process.
+\! gpconfig -c shared_preload_libraries -v 'gp_inject_fault';
```

Comment:
Does the test work without that? I run locally without `gpconfig ` and cluster restart and everything passes.

Reply:
inject fault may be applied by `create extension gp_inject_fault`

Reply:
>> Does the test work without that? I run locally without gpconfig  and cluster restart and everything passes.
Indeed, will remove it.

>> Can we be confident about pg_sleep (1) is enough to make the test stable?
It should be enough I think. BackgroundWriterMain usually sleeps for default 200ms (unless somebody changes this GUC).  And to catch the problem by this test, we do not need to wait for all data to be flushed by bg writer process - we just need it to write some data, not necessarily all.
BackgroundWriterMain can go to hibernation for 10 seconds, but when a backend starts using buffers again, it will wake up BackgroundWriterMain by setting the latch. So it looks there is no need to extend the delay.